app/models/time_period.rb

```diff
@@ -53,10 +53,18 @@ def previous_time_period
     def create_time_period
       return if current_time_period.present?
 
-      start_week = ENV['START_WEEK_DAY']&.to_sym || :monday
-      start_date = Date.current.beginning_of_week(start_week)
+      start_date = Date.current.beginning_of_week(start_date_wday)
       end_date = start_date + 6.days
-      TimePeriod.create(start_date: start_date, end_date: end_date)
+      due_date = (start_date..end_date).find { |date| date.strftime('%A').downcase.to_sym == due_date_wday }
+      TimePeriod.create!(start_date:, end_date:, due_date:)
+    end
+
+    def start_date_wday
+      ENV['START_WEEK_DAY']&.to_sym || :monday
+    end
+
+    def due_date_wday
```

Comment:
`day_that_week_starts` and `day_to_send_invites` might be more clear and accurate.